app/services/service_service.js

```diff
@@ -61,26 +61,26 @@ function getGatewayAccounts (gatewayAccountIds, correlationId) {
  * @returns {Promise<Service>} the updated service
  */
 function updateServiceName (serviceExternalId, serviceName, serviceNameCy, correlationId) {
-  return new Promise(function (resolve, reject) {
-    if (!serviceExternalId) reject(new Error(`argument: 'serviceExternalId' cannot be undefined`))
-    getAdminUsersClient({ correlationId }).updateServiceName(serviceExternalId, serviceName, serviceNameCy)
-      .then(result => {
-        const gatewayAccountIds = lodash.get(result, 'gateway_account_ids', [])
-        // Update gateway account service names
-        if (gatewayAccountIds.length <= 0) {
-          return resolve(new Service(result))
-        } else {
-          const accounts = lodash.partition(gatewayAccountIds, id => isADirectDebitAccount(id))
-          const gatewayAccountId = accounts[1]
-          if (gatewayAccountId && gatewayAccountId.length > 0 && !isADirectDebitAccount(gatewayAccountId)) {
+  if (!serviceExternalId) {
+    return Promise.reject(new Error(`argument: 'serviceExternalId' cannot be undefined`))
+  }
+  return getAdminUsersClient({ correlationId }).updateServiceName(serviceExternalId, serviceName, serviceNameCy)
+    .then(result => {
+      // Update gateway account service names in connector
+      const gatewayAccountIds = lodash.get(result, 'gateway_account_ids', [])
+      if (gatewayAccountIds.length <= 0) {
+        return new Service(result)
+      } else {
+        return Promise.all(gatewayAccountIds.map(gatewayAccountId => {
```

Comment:
This `map` needs to return a result in order to wait on the result. Right now it looks like this will just `Promise.all[undefined]` which will just skip.

I.e 
```js 
map((id) => { 
  return connectorClient.patchServiceName(id)
}
```